integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/util/SecondaryIndexUtil.scala

```diff
@@ -310,20 +284,22 @@ object SecondaryIndexUtil {
   /**
    * This method deletes the old index files or merge index file after data files merge
    */
-  private def deleteOldIndexOrMergeIndexFiles(
+  private def collectOldIndexOrMergeIndexFiles(
       factTimeStamp: Long,
       validSegments: util.List[Segment],
       indexCarbonTable: CarbonTable): Unit = {
+    val oldIndexAndMergeIndexFiles = new util.ArrayList[String]
     // delete the index/merge index carbonFile of old data files
     validSegments.asScala.foreach { segment =>
       SegmentFileStore.getIndexFilesListForSegment(segment, indexCarbonTable.getTablePath)
         .asScala
         .foreach { indexFile =>
           if (DataFileUtil.getTimeStampFromFileName(indexFile).toLong < factTimeStamp) {
-            FileFactory.getCarbonFile(indexFile).delete()
+            oldIndexAndMergeIndexFiles.add(indexFile)
           }
         }
     }
+    SegmentIndexFileStore.setOldIndexAndMergeIndexFiles(oldIndexAndMergeIndexFiles)
```

Comment:
please check if there is any test case for merge data files and index files check, if not add and assert it wll contain old index files before clean and doesnt contain after clean files

Reply:
modified one existing test case and added one to check data file merge without enabling index merge property.